### ISSUE TITLE
Screen Lock - Weird Issues (Blanking after pressing a keybind) and Unresponsivness

### ISSUE THREAD

**Issue by SpyderGamer** [linuxmint/cinnamon-screensaver#451]
### Distribution

Linux Mint 21.3

### Package version

Cinnamon 6.0.2

### Graphics hardware in use

AMD ATI 03:00.0 Barcelo

### Frequency

Always

### Bug description

Lock screen unresponsive and slow/sluggish. Keybindings (Brightness up/down, Volume up/down, Caps lock, Num lock) cause it to go blank for a while.

### Steps to reproduce

1. Switch to the desktop without any app visible on the screen (Any programs such as Firefox, Terminal, or Discord must be minimized. `Super + D` can also be used).
2. Lock the screen using the keyboard shortcut `Ctrl + Alt + L` or wake up after suspend.
3. The lock screen animations become very slow and unresponsive (Try entering an incorrect password, the confirmation bar animation runs at what seems like 10 FPS). Though everything is functional and I can still log in. Additionally, when I press any function keys/keybindings while on the lock screen (Brightness up/down, Volume up/down, Caps lock, Num lock), the screen becomes blank for a second or two then goes back to normal.
4. However, after logging in, keep an app maximized and visible then attempt to lock the screen again. This time, the animations are fluent and the lock screen is much more responsive. And pressing the function keys/keybdings don't blank the screen.

### Expected behavior

The lock screen should be smooth and responsive whether or not an app was visible on the screen before locking. And pressing function keys/keybindings should not blank the screen.

### Additional information

Heres a similar issues: https://github.com/linuxmint/cinnamon-screensaver/issues/450, https://github.com/linuxmint/cinnamon/issues/11632

System info:
```
System:
  Kernel: 5.15.0-91-generic x86_64 bits: 64 compiler: gcc v: 11.4.0 Desktop: Cinnamon 6.0.4
    tk: GTK 3.24.33 wm: muffin vt: 7 dm: LightDM 1.30.0 Distro: Linux Mint 21.3 Virginia
    base: Ubuntu 22.04 jammy
Machine:
  Type: Laptop System: HP product: HP Pavilion Laptop 15-eh3xxx v: N/A
    serial: <superuser required> Chassis: type: 10 serial: <superuser required>
  Mobo: HP model: 8BC7 v: 68.20 serial: <superuser required> UEFI: AMI v: F.03 date: 07/04/2023
Battery:
  ID-1: BAT0 charge: 29.5 Wh (84.3%) condition: 35.0/35.0 Wh (100.0%) volts: 11.9 min: 11.3
    model: HP Primary type: Li-ion serial: <filter> status: Discharging cycles: 190
CPU:
  Info: 8-core model: AMD Ryzen 7 7730U with Radeon Graphics bits: 64 type: MT MCP smt: enabled
    arch: Zen 3 rev: 0 cache: L1: 512 KiB L2: 4 MiB L3: 16 MiB
  Speed (MHz): avg: 3753 high: 4541 min/max: 1600/2000 boost: enabled cores: 1: 3631 2: 3632
    3: 3632 4: 3632 5: 3640 6: 3633 7: 3633 8: 3630 9: 3717 10: 3656 11: 3633 12: 3625 13: 3632
    14: 3643 15: 4541 16: 4541 bogomips: 63879
  Flags: avx avx2 ht lm nx pae sse sse2 sse3 sse4_1 sse4_2 sse4a ssse3 svm
Graphics:
  Device-1: AMD Barcelo vendor: Hewlett-Packard driver: amdgpu v: kernel pcie: speed: 8 GT/s
    lanes: 16 ports: active: eDP-1 empty: DP-1,HDMI-A-1 bus-ID: 03:00.0 chip-ID: 1002:15e7
    class-ID: 0300
  Device-2: Chicony HP Wide Vision HD Camera type: USB driver: uvcvideo bus-ID: 1-3:2
    chip-ID: 04f2:b760 class-ID: fe01 serial: <filter>
  Display: x11 server: X.Org v: 1.21.1.4 driver: X: loaded: amdgpu,ati
    unloaded: fbdev,modesetting,vesa gpu: amdgpu display-ID: :0 screens: 1
  Screen-1: 0 s-res: 1920x1080 s-dpi: 96 s-size: 508x285mm (20.0x11.2") s-diag: 582mm (22.9")
  Monitor-1: eDP res: 1920x1080 hz: 60 dpi: 142 size: 344x193mm (13.5x7.6") diag: 394mm (15.5")
  OpenGL: renderer: RENOIR (renoir LLVM 15.0.7 DRM 3.42 5.15.0-91-generic)
    v: 4.6 Mesa 23.0.4-0ubuntu1~22.04.1 direct render: Yes
Audio:
  Device-1: AMD Renoir Radeon High Definition Audio vendor: Hewlett-Packard driver: snd_hda_intel
    v: kernel pcie: speed: 8 GT/s lanes: 16 bus-ID: 03:00.1 chip-ID: 1002:1637 class-ID: 0403
  Device-2: AMD Raven/Raven2/FireFlight/Renoir Audio Processor vendor: Hewlett-Packard
    driver: snd_rn_pci_acp3x v: kernel pcie: speed: 8 GT/s lanes: 16 bus-ID: 03:00.5
    chip-ID: 1022:15e2 class-ID: 0480
  Device-3: AMD Family 17h HD Audio vendor: Hewlett-Packard driver: snd_hda_intel v: kernel
    pcie: speed: 8 GT/s lanes: 16 bus-ID: 03:00.6 chip-ID: 1022:15e3 class-ID: 0403
  Sound Server-1: ALSA v: k5.15.0-91-generic running: yes
  Sound Server-2: PulseAudio v: 15.99.1 running: yes
  Sound Server-3: PipeWire v: 0.3.48 running: yes
Network:
  Device-1: MEDIATEK MT7921 802.11ax PCI Express Wireless Network Adapter vendor: AzureWave
    driver: mt7921e v: kernel pcie: speed: 5 GT/s lanes: 1 bus-ID: 01:00.0 chip-ID: 14c3:7961
    class-ID: 0280
  IF: wlo1 state: up mac: <filter>
Bluetooth:
  Device-1: IMC Networks Wireless_Device type: USB driver: btusb v: 0.8 bus-ID: 1-4:3
    chip-ID: 13d3:3567 class-ID: e001 serial: <filter>
  Report: hciconfig ID: hci0 rfk-id: 0 state: up address: <filter> bt-v: 3.0 lmp-v: 5.2
    sub-v: 2202 hci-v: 5.2 rev: 915
Drives:
  Local Storage: total: 476.94 GiB used: 82.02 GiB (17.2%)
  ID-1: /dev/nvme0n1 vendor: Intel model: SSDPEKNU512GZH size: 476.94 GiB speed: 31.6 Gb/s
    lanes: 4 type: SSD serial: <filter> rev: HPS1 temp: 32.9 C scheme: GPT
Partition:
  ID-1: / size: 376.43 GiB used: 67.21 GiB (17.9%) fs: ext4 dev: /dev/nvme0n1p2
  ID-2: /boot/efi size: 511 MiB used: 4.2 MiB (0.8%) fs: vfat dev: /dev/nvme0n1p1
Swap:
  ID-1: swap-1 type: file size: 2 GiB used: 0 KiB (0.0%) priority: -2 file: /swapfile
USB:
  Hub-1: 1-0:1 info: Hi-speed hub with single TT ports: 4 rev: 2.0 speed: 480 Mb/s
    chip-ID: 1d6b:0002 class-ID: 0900
  Device-1: 1-2:4 info: Logitech M-BJ58/M-BJ69 Optical Wheel Mouse type: Mouse
    driver: hid-generic,usbhid interfaces: 1 rev: 2.0 speed: 1.5 Mb/s power: 98mA chip-ID: 046d:c00e
    class-ID: 0301
  Device-2: 1-3:2 info: Chicony HP Wide Vision HD Camera type: Video driver: uvcvideo
    interfaces: 3 rev: 2.0 speed: 480 Mb/s power: 500mA chip-ID: 04f2:b760 class-ID: fe01
    serial: <filter>
  Device-3: 1-4:3 info: IMC Networks Wireless_Device type: Bluetooth driver: btusb interfaces: 3
    rev: 2.1 speed: 480 Mb/s power: 100mA chip-ID: 13d3:3567 class-ID: e001 serial: <filter>
  Hub-2: 2-0:1 info: Super-speed hub ports: 2 rev: 3.1 speed: 10 Gb/s chip-ID: 1d6b:0003
    class-ID: 0900
  Hub-3: 3-0:1 info: Hi-speed hub with single TT ports: 4 rev: 2.0 speed: 480 Mb/s
    chip-ID: 1d6b:0002 class-ID: 0900
  Device-1: 3-4:2 info: Elan Micro ELAN:ARM-M4 type: <vendor specific> driver: N/A interfaces: 1
    rev: 2.0 speed: 12 Mb/s power: 100mA chip-ID: 04f3:0c00 class-ID: 0000
  Hub-4: 4-0:1 info: Super-speed hub ports: 2 rev: 3.1 speed: 10 Gb/s chip-ID: 1d6b:0003
    class-ID: 0900
Sensors:
  System Temperatures: cpu: 41.0 C mobo: N/A gpu: amdgpu temp: 43.0 C
  Fan Speeds (RPM): N/A
Repos:
  Packages: 2416 apt: 2368 flatpak: 48
  No active apt repos in: /etc/apt/sources.list
  Active apt repos in: /etc/apt/sources.list.d/official-package-repositories.list
    1: deb http: //packages.linuxmint.com virginia main upstream import backport
    2: deb http: //archive.ubuntu.com/ubuntu jammy main restricted universe multiverse
    3: deb http: //archive.ubuntu.com/ubuntu jammy-updates main restricted universe multiverse
    4: deb http: //archive.ubuntu.com/ubuntu jammy-backports main restricted universe multiverse
    5: deb http: //security.ubuntu.com/ubuntu/ jammy-security main restricted universe multiverse
  Active apt repos in: /etc/apt/sources.list.d/vscode.list
    1: deb [arch=amd64,arm64,armhf] http: //packages.microsoft.com/repos/code stable main
Info:
  Processes: 418 Uptime: 14h 1m wakeups: 10730 Memory: 14.96 GiB used: 4.63 GiB (30.9%)
  Init: systemd v: 249 runlevel: 5 Compilers: gcc: 11.4.0 alt: 11/12
  Client: Unknown python3.10 client inxi: 3.3.13
```

<br>

Last couple of lines of .xsession-errors:
```
cinnamon-session[1375]: WARNING: t+2207.32145s: Detected that screensaver has appeared on the bus
cinnamon-session[1375]: WARNING: t+2266.51051s: Detected that screensaver has left the bus
cinnamon-session[1375]: WARNING: t+2380.63650s: Detected that screensaver has appeared on the bus
cinnamon-session[1375]: WARNING: t+2430.34907s: Detected that screensaver has left the bus
/usr/share/cinnamon/cinnamon-settings/cinnamon-settings.py:444: DeprecationWarning: Gtk.Window.set_wmclass is deprecated
  self.window.set_wmclass(wm_class, wm_class)
cinnamon-session[1375]: WARNING: t+2456.32848s: Detected that screensaver has appeared on the bus
cinnamon-session[1375]: WARNING: t+2489.52870s: Detected that screensaver has left the bus
cinnamon-session[1375]: WARNING: t+2500.77827s: Detected that screensaver has appeared on the bus
cinnamon-session[1375]: WARNING: t+2576.43648s: Detected that screensaver has left the bus
cinnamon-session[1375]: WARNING: t+2671.24873s: Detected that screensaver has appeared on the bus
cinnamon-session[1375]: WARNING: t+2704.89141s: Detected that screensaver has left the bus
cinnamon-session[1375]: WARNING: t+2821.91602s: Detected that screensaver has left the bus

** (csd-automount:1501): WARNING **: 10:51:06.788: Can't call GetActive() on the ScreenSaver object: Cannot invoke method; proxy is for the well-known name org.cinnamon.ScreenSaver without an owner, and proxy was constructed with the G_DBUS_PROXY_FLAGS_DO_NOT_AUTO_START flag
cinnamon-session[1375]: WARNING: t+3262.90069s: Detected that screensaver has appeared on the bus
cinnamon-session[1375]: WARNING: t+3306.35126s: Detected that screensaver has left the bus

(cinnamon:6114): St-WARNING **: 10:59:27.527: parse_shadow_property: default case

(cinnamon:6114): St-WARNING **: 10:59:27.527: parse_shadow_property: default case

(cinnamon:6114): St-WARNING **: 10:59:27.527: parse_shadow_property: default case

(cinnamon:6114): St-WARNING **: 10:59:27.527: parse_shadow_property: default case

(cinnamon:6114): St-WARNING **: 10:59:27.527: parse_shadow_property: default case

(cinnamon:6114): St-WARNING **: 10:59:27.527: parse_shadow_property: default case

(cinnamon:6114): St-WARNING **: 10:59:27.527: parse_shadow_property: default case

(cinnamon:6114): St-WARNING **: 10:59:27.527: parse_shadow_property: default case
cinnamon-session[1375]: WARNING: t+3514.16973s: Detected that screensaver has appeared on the bus

(cinnamon:6114): mutter-WARNING **: 11:02:45.376: Called enable_unredirect_for_display while unredirection is enabled.
cinnamon-session[1375]: WARNING: t+3551.08945s: Detected that screensaver has left the bus
```

**Comment by mtwebster** [comment]
Hi are you using any sort of display scaling? Fractional or otherwise.

Also, can you start cinnamon-screensaver from a terminal this way:
```
cinnamon-screensaver --hold --debug
```

And paste the output when this happens.

Thanks

**Comment by SpyderGamer** [comment]
> Hi are you using any sort of display scaling? Fractional or otherwise.
> 
> Also, can you start cinnamon-screensaver from a terminal this way:
> 
> ```
> cinnamon-screensaver --hold --debug
> ```
> 
> And paste the output when this happens.
> 
> Thanks

Nope, there is no scaling enabled:
![image](https://github.com/linuxmint/cinnamon-screensaver/assets/85440857/5d7d624f-3066-4669-8757-ab568a38cdd1)

Here's the output of the command (Changed my username):
```
Loading AccountsService
Fractional scaling active: False
Trying to connect to logind...
Starting screensaver...
Debug mode active
Cinnamon Screensaver support not found in current theme - adding some...
service: 'GetActive' received, poking application.
service: 'GetActive' received, poking application.
org.gnome.ScreenSaver is gone from the session bus
org.mate.ScreenSaver is gone from the session bus
service: 'GetActive' received, poking application.
login client: found session path for user 'myUserName' (session_id: c2): /org/freedesktop/login1/session/c2
Successfully using logind
AccountsService ready
^C
```

Thanks for your reply.

**Comment by SpyderGamer** [comment]
This issue has been fixed with a recent update to Cinnamon 6.0.3. Thanks to the contributors who fixed this!

**Comment by SpyderGamer** [comment]
This issue has been fixed with a recent update to Cinnamon 6.0.3. Thanks to the contributors who fixed this!

**Comment by Perlover** [comment]
The same result - I have not this problem after update now
